File: test/org/jetbrains/android/dsl/tests/functional/ComplexListenerClassTest.java
package org.jetbrains.android.dsl.tests.functional;

import org.jetbrains.android.dsl.BaseGeneratorProps;
import org.jetbrains.android.dsl.Subsystem;
import org.testng.annotations.BeforeMethod;
import org.testng.annotations.Test;

public class ComplexListenerClassTest extends BaseFunctionalTest {
    private final String testDataFile = "org/jetbrains/android/dsl/tests/functional/ComplexListenerClassTest.kt";

    @Override
    @BeforeMethod
    public void setUp() throws Exception {
        super.setUp();
    }

    protected void initSettings(BaseGeneratorProps settings) {
        settings.setGenerateComplexListenerClasses(true);
    }

    @Test
    public void testComplexListeners() throws Exception {
        runFunctionalTest(testDataFile, Subsystem.LISTENERS);
    }
}

File: test/org/jetbrains/android/dsl/functional/ComplexListenerClassTest.kt
package org.jetbrains.android.dsl.functional
import org.jetbrains.android.dsl.*
import org.testng.annotations.Test
public class ComplexListenerClassTest : AbstractFunctionalTest() {
    private val testDataFile = "ComplexListenerClassTest.kt"
    override fun initSettings(settings: BaseGeneratorConfiguration) {
        settings.files.add(KoanFile.LISTENERS)
        settings.tunes.add(ConfigurationTune.COMPLEX_LISTENER_CLASSES)
    }
    [Test] public fun testComplexListenerClassTestFor19() {
        runFunctionalTest(testDataFile, KoanFile.LISTENERS, "19")
    }
    [Test] public fun testComplexListenerClassTestFor19s() {
        runFunctionalTest(testDataFile, KoanFile.LISTENERS, "19s")
    }
}
